### ISSUE TITLE
p-fileupload's upload & cancel buttons are clickable even when no file is selected

### ISSUE THREAD

**Issue by yasstou** [primefaces/primeng#7753]
**I'm submitting a ...**  (check one with "x")
```
[x ] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```
**Current behavior**

After 7.0.3 in fileupload.ts, the code for the upload & cancel buttons was changed from this : 
```
<button *ngIf="!auto&&showUploadButton" type="button" [label]="uploadLabel" icon="pi pi-upload" pButton (click)="upload()" [disabled]="!hasFiles()"></button>
<button *ngIf="!auto&&showCancelButton" type="button" [label]="cancelLabel" icon="pi pi-times" pButton (click)="clear()" [disabled]="!hasFiles()"></button>
```
to this : 
```
<p-button *ngIf="!auto&&showUploadButton" type="button" [label]="uploadLabel" icon="pi pi-upload" (click)="upload()" [disabled]="!hasFiles()"></p-button>
<p-button *ngIf="!auto&&showCancelButton" type="button" [label]="cancelLabel" icon="pi pi-times" (click)="clear()" [disabled]="!hasFiles() || uploading"></p-button>
```
This change made the buttons clickable even when disabled and also broke the css in primeng.css because now we the `<button class="ui-button"></button>` is inside `<p-button></p-button>` which messes up button margins and prevents them from taking the full width in mobile view.
<pre>
@media (max-width: 40em)
.ui-fileupload-buttonbar .ui-button:last-child {
    margin-bottom: 0;
}
@media (max-width: 40em)
.ui-fileupload-buttonbar .ui-button {
    display: block;
    width: 100%;
    margin-right: 0;
    <s>margin-bottom: .25em;</s>
}
</pre>

**Expected behavior**

The upload & cancel buttons should remain disabled until a file is selected and should take the full width when in mobile view.

**Minimal reproduction of the problem with instructions**
https://stackblitz.com/edit/github-upkjow-dn17go

* **Angular version:** 7.2.X

* **PrimeNG version:** 7.1.3

* **Browser:** [all]

**Comment by ewinchen** [comment]
meet this issue too, even the official document demo work like this
https://www.primefaces.org/primeng/#/fileupload

**Comment by runderworld** [comment]
broken in [7.x](https://www.primefaces.org/primeng-7.1.3/#/fileupload), seemingly fixed in [8.x](https://www.primefaces.org/primeng/#/fileupload).

**Comment by yigitfindikli** [comment]
I’m unable to replicate with the latest PrimeNG. If the issue still persists with the new version please create a new issue with our stackblitz template.
https://stackblitz.com/github/primefaces/primeng-issue-template

**Comment by andrewleedev** [comment]
This fix is verified in 8.0.2.
FileLimit on FileUpload #[7906](https://github.com/primefaces/primeng/issues/7906)
Relevant pull request: [7981](https://github.com/primefaces/primeng/pull/7981)
